src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K,V>[] table;
+    private int size = 0;
+    private int threshold;
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int initialCapacity) {
+        if (initialCapacity <= 0) {
+            throw new IllegalArgumentException("Illegal initial capacity: "
+                                               + initialCapacity);
+        }
+        int capacity = DEFAULT_INITIAL_CAPACITY;
+        while (capacity < initialCapacity) {
+            capacity <<= 1;
+        }
+        table = (Node<K,V>[]) new Node[capacity];
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    static class Node<K, V> {
+        int hash;
```

Comment:
do we need to keep int hash in this Node class?

Reply:
Agree, removed